addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3315,7 +3315,8 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
         $weight.find('b').text(`${(weight / 1024).toFixed(1)} kb`);
         $weight.removeClass('d-none');
         img.classList.add('o_modified_image_to_save');
-        return loadImage(dataURL, img);
+        await loadImage(dataURL, img);
+        return this._applyImage(img);
```

Comment:
Nearly good but now this fix is not entirely stable: before your commit, `_applyOptions` return the result of `loadImage`, which is a `Promise` instance resolving to an `HTMLImageElement`. Now, it returns a `Promise` instance which does not resolve to anything (as the function is `async` here, its result is always a Promise, but `_applyImage` has not return value).

Change these two lines like this:
```suggestion
        const loadedImg = await loadImage(dataURL, img);
        this._applyImage(loadedImg);
        return loadedImg;
```
```